kubectl-sniff:init at 1.6.2

behoof4mind opened the pull request:
## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

behoof4mind committed:
kubectl-sniff:init at 1.6.2

fix nixfmt-check

Revert "fix nixfmt-check"

This reverts commit a7ad539722a34b6a936a2c0203d655cde29afb54.

fix format by nixpkgs-fmt

fix format by nixfmt-rfc-style

roll-back mainteiners-list.nix

apply changes requested by @gador

gador commented on maintainers/maintainer-list.nix:
```diff
@@ -2,18 +2,18 @@
    handle = {
-     # Required
-     name = "Your name";
+   # Required
```
Do not change unrelated fileds in `maintainer-list.nix`

gador commented on maintainers/maintainer-list.nix:
```diff
@@ -2424,6 +2424,13 @@
     github = "beezow";
     githubId = 42082156;
   };
+  behoof4mind = {
+    name = "Denis Lavrushko";
+    email = "dlavrushko@protonmail.com";
+    github = "behoof4mind";
+    githubId = 9162748;
+    keys = [ { fingerprint = "F4B6 7C35 BD08 6C48 32C4  3140 4B5E 6DEB F81C EFB6"; } ];
+  };
```
Please use a seperate commit `maintainers: add behoof4mind`

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+
+  src = fetchurl {
+    url = "https://github.com/${owner}/${pname}/releases/download/v${version}/ksniff.zip";
+    sha256 = "sha256-xZtcnqhNbLdxCW8SRskZtxOJ+dQjToWPSSkgiVflYf0=";
```
```suggestion
    hash = "sha256-xZtcnqhNbLdxCW8SRskZtxOJ+dQjToWPSSkgiVflYf0=";
```

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+}:
+
+stdenv.mkDerivation rec {
+  owner = "eldadru";
```
```suggestion
```

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+  version = "1.6.2";
+
+  src = fetchurl {
+    url = "https://github.com/${owner}/${pname}/releases/download/v${version}/ksniff.zip";
```
```suggestion
    url = "https://github.com/eldadru/ksniff/releases/download/v${version}/ksniff.zip";
```

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+  installPhase = ''
+    mkdir -p $out/bin
+    cp $out/static-tcpdump $out/bin/static-tcpdump
+    case ${stdenv.hostPlatform.system} in
+      x86_64-linux)
+        cp $out/kubectl-sniff $out/bin/kubectl-sniff
+        ;;
+      x86_64-darwin)
+        cp $out/kubectl-sniff-darwin $out/bin/kubectl-sniff
+        ;;
+      aarch64-darwin)
+        cp $out/kubectl-sniff-darwin-arm64 $out/bin/kubectl-sniff
+        ;;
+      *)
+        echo "Unsupported system: ${stdenv.hostPlatform.system}"
+        exit 1
+        ;;
```
Are these artifacts precompiled? If so, can we not build from source? 
Also, should we not patch these then with `autoPatchelfHook`?

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+        exit 1
+        ;;
+    esac
+    mv $out/LICENSE $out/LICENSE.ksniff
```
```suggestion
```

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+    mv $out/LICENSE $out/LICENSE.ksniff
+  '';
+
+  meta = with lib; {
```
```suggestion
  meta = {
```

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+  meta = with lib; {
+    description = "Kubernetes CLI plugin for live capture of network traffic between pods";
+    homepage = "https://github.com/eldadru/ksniff";
+    license = licenses.mit;
```
```suggestion
    license = lib.licenses.mit;
```

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+    description = "Kubernetes CLI plugin for live capture of network traffic between pods";
+    homepage = "https://github.com/eldadru/ksniff";
+    license = licenses.mit;
+    maintainers = with maintainers; [ maintainers.behoof4mind ];
```
```suggestion
    maintainers = with lib.maintainers; [ behoof4mind ];
```

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,53 @@
+    homepage = "https://github.com/eldadru/ksniff";
+    license = licenses.mit;
+    maintainers = with maintainers; [ maintainers.behoof4mind ];
+    platforms = platforms.linux ++ platforms.darwin;
```
```suggestion
    platforms = with lib.platforms; linux ++ darwin;
```

gador reviewed (changes requested):
Thanks for your contribution! There are a few changes I'd like to see before we can merge this. Feel free to ping me for questions.

behoof4mind committed:
maintainers: add behoof4mind

gador commented on pkgs/applications/networking/cluster/ksniff/default.nix:
```diff
@@ -0,0 +1,51 @@
+    homepage = "https://github.com/eldadru/ksniff";
+    license = lib.licenses.mit;
+    maintainers = with lib.maintainers; [ behoof4mind ];
+    platforms = lib.platforms; linux ++ darwin;
```
```suggestion
    platforms = with lib.platforms; linux ++ darwin;
```

behoof4mind commented:
@gador, thank you for your review.
Sorry for delay, but now seems like everything is updated properly.

This line is here https://github.com/NixOS/nixpkgs/pull/331587/files#diff-a74b8276623cfd33bf9cfe7fc5f1872bf8fd3f679b704b4ac052a5cf6b8231a0R25, because for me LICENSE file conflict with another one from `php81Packages.composer`

gador commented:
@behoof4mind Thanks for addressing the comments! I'm still wondering about this though:
> Are these artifacts precompiled? If so, can we not build from source?
> Also, should we not patch these then with autoPatchelfHook?

https://github.com/NixOS/nixpkgs/pull/331587#discussion_r1701343975

behoof4mind commented:
> @behoof4mind Thanks for addressing the comments! I'm still wondering about this though:
> 
> > Are these artifacts precompiled? If so, can we not build from source?
> > Also, should we not patch these then with autoPatchelfHook?
> 
> [#331587 (comment)](https://github.com/NixOS/nixpkgs/pull/331587#discussion_r1701343975)

Oh, man. You're right.
I missed the thing that static-tcpdump is precompiled :/

I will make changes, so the dependancy will build by hook 👍

behoof4mind committed:
Merge branch 'NixOS:master' into master
